### PR TITLE
refactor/decorator ordering

### PR DESCRIPTION
**Pull request by b-butler**
## Description
Remove show-traceback completely from CLI and config. Deprecated assigning conditions before registering operations. Update test decorator ordering. Update documentation examples to new decorator ordering.

## Motivation and Context

Fixes a discussion on Slack about `with_job=True` invaliding conditions already specified as we store conditions in a `dict` where function hash is used for the key and `with_job=True` wraps the function. This is also desirable as it requires a more Pythonic order to the decorators and allows potential refactoring in the future to simplify the code.

## Checklist:

- [x] I am familiar with the [Contributing Guidelines](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [Contributor Agreement](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [x] The [package documentation](https://github.com/glotzerlab/signac-flow/tree/master/doc) and [framework documentation](https://docs.signac.io/) in [signac-docs](https://github.com/glotzerlab/signac-docs) are up to date with these changes.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt) and added any related issue and pull request numbers for future reference.

### CODE REVIEW

**Review by csadorf — APPROVED** [review]
Great! Thanks a lot!

**Review by shihkual — APPROVED** [review]
Looks good to me!

File: flow/project.py [hunk 1]
```diff
@@ -1287,10 +1286,20 @@ def hi_all(*jobs):
...
             def __call__(self, func):
+                # Have to traverse the mro to ensure that func is already an operation and that
+                # self.condition is not.
                 operation_functions = [
-                    operation[1]
-                    for operation in self._parent_class._collect_operations()
+                    func for name, func in self._parent_class._collect_operations()
                 ]
+                if func not in operation_functions:
+                    _deprecated_warning(
```
Comment (bdice) [inline flow/project.py:1295]: This will raise instead of warn in the future, right? In other words, we still have to do this check in the future.

**Review by bdice — APPROVED** [review]

File: flow/project.py (hunk 1 above)
Comment (b-butler) [reply flow/project.py:1294]: Yes or something like it.

**Pull request merged**